pkg/rc/replication_controller.go

```diff
@@ -633,81 +661,277 @@ func (rc *replicationController) unschedule(txn *auditingTransaction, node types
 	return nil
 }
 
-func (rc *replicationController) transferNodes(ineligibleCurrent []types.NodeName) error {
-	var err error
-	if rc.AllocationStrategy == fields.CattleStrategy {
-		err = rc.transferCattleNodes(ineligibleCurrent)
+func (rc *replicationController) transferNodes(ineligible []types.NodeName) error {
+	if rc.AllocationStrategy != fields.CattleStrategy {
+		errMsg := fmt.Sprintf("Non-cattle RC has scheduled %d ineligible nodes: %s", len(ineligible), ineligible)
+		err := rc.alerter.Alert(rc.alertInfo(errMsg), alerting.LowUrgency)
+		if err != nil {
+			rc.logger.WithError(err).Errorln("Unable to send alert")
+		}
+		// We don't return an error here because this is the current behavior for non-cattle
+		// ineligible nodes: only page
+		return nil
+	}
+
+	status, _, err := rc.rcStatusStore.Get(rc.ID())
+	if err != nil && !statusstore.IsNoStatus(err) {
+		return err
+	}
+
+	var newNode types.NodeName
+	if status.NodeTransfer == nil {
+		newNode, err = rc.updateAllocations(ineligible)
+		if err != nil {
+			return err
+		}
+
+		err := rc.scheduleWithSession(newNode)
+		if err != nil {
+			// Different RC may have already taken over the new node, abort transfer
+			rc.quitTransfer()
+			return err
+		}
 	} else {
-		err = rc.transferPetNodes(ineligibleCurrent)
+		newNode = status.NodeTransfer.NewNode
 	}
-	if err != nil {
-		return util.Errorf("Unable to transfer nodes: %s", err)
+
+	if rc.isWatchingHealth {
+		return nil
 	}
 
+	rc.nodeTransfer = nodeTransfer{
+		newNode: newNode,
+		quit:    make(chan struct{}),
+	}
+
+	go rc.watchHealth()
```

Comment:
`watchHealth()` does some stuff and then calls `finishTransfer()` It might be clearer if you called something like `rc.doNodeTransfer()` here which is just something like 

```
func doNodeTransfer() {
  watchHealth()
  finishTransfer()
}
``` (obviously that's pseudocode)

That I think more clearly shows the intent here and also it means you don't have to call `finishTransfer` in multiple places from `watchHealth`